### ISSUE TITLE
Bug in calling dgeevx from LAPACK

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#928]
**Modified by dietmarw on 12 Dec 2012 09:45 UTC**
Currently in `dgeevx` the LAPACK routine is called as
```
external "Fortran 77" dgeevx("B","V","V","E",n,AS,n,alphaReal,alphaImag,lEigenVectors,n,rEigenVectors,n,ilo,ihi,scale,abnrm,rconde,rcondv,work,lwork,info)annotation(Library={"lapack"});
```

Here the parameter `iwork` is missing according to the LAPACK spec of `DGGEVX`:
```
      SUBROUTINE DGEEVX( BALANC, JOBVL, JOBVR, SENSE, N, A, LDA, WR, WI,
     $                   VL, LDVL, VR, LDVR, ILO, IHI, SCALE, ABNRM,
     $                   RCONDE, RCONDV, WORK, LWORK, IWORK, INFO )

```
Because `dgeevx` is called with `sense="E"`, `iwork` is not referenced and can be introduced as
```
    protected
        Integer iwork[1];
```

----

**Reported by bastian on 12 Dec 2012 09:19 UTC**
Currently in `dgeevx` the LAPACK routine is called as
```
external "Fortran 77" dgeevx("B","V","V","E",n,AS,n,alphaReal,alphaImag,lEigenVectors,n,rEigenVectors,n,ilo,ihi,scale,abnrm,rconde,rcondv,work,lwork,info)annotation(Library={"lapack"});
```

Here the parameter `iwork` is missing according to the LAPACK spec of `DGGEVX`:
```
      SUBROUTINE DGEEVX( BALANC, JOBVL, JOBVR, SENSE, N, A, LDA, WR, WI,
     $                   VL, LDVL, VR, LDVR, ILO, IHI, SCALE, ABNRM,
     $                   RCONDE, RCONDV, WORK, LWORK, IWORK, INFO )

```
Because `dgeevx` is called with `sense="E"`, `iwork` is not referenced and can be introduced as
```
    protected
        Real iwork[1];
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/928

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 12 Dec 2012 09:45 UTC**
Fixed the description.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 12 Dec 2012 10:07 UTC**
Fixed in 8ec5bca2532689737b0b56db94e77c75cb4ac862
